### PR TITLE
ttl: disable ttl job when recover/flashback table/database/cluster

### LINKED ISSUE #40265: Make TTL compatible with `RECOVER` and `FLASHBACK`

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #40265

### What is changed and how it works?

According to the product design, the ttl job should be disabled when recover or flashback a table. The behavior in this PR is:

- flashback/recover table: force set table option `TTL_ENABLE='OFF'`
- flashback database: force set all table options in the cluster `TTL_ENABLE='OFF'`
- flashback cluster to timestamp: force set global system var `TTL_JOB_ENABLE='OFF'`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
ttl: disable ttl job when recover/flashback table/database/cluster
```

### CODE REVIEW

File: ddl/cluster.go [hunk 1]
```diff
@@ -696,8 +703,9 @@ func finishFlashbackCluster(w *worker, job *model.Job) error {
 	var pdScheduleValue map[string]interface{}
 	var autoAnalyzeValue, readOnlyValue string
 	var gcEnabled bool
+	ttlEnabled := true
 
-	if err := job.DecodeArgs(&flashbackTS, &pdScheduleValue, &gcEnabled, &autoAnalyzeValue, &readOnlyValue, &lockedRegions, &startTS, &commitTS); err != nil {
```
Comment (Defined2014) [inline ddl/cluster.go:700]: Should store session variable in `model.StateNone` state. But one question here, what will happened when ttlEnabled during flashback?

File: ddl/cluster.go [hunk 2]
```diff
@@ -176,6 +180,9 @@ func checkAndSetFlashbackClusterInfo(sess sessionctx.Context, d *ddlCtx, t *meta
 	if err = setTiDBSuperReadOnly(d.ctx, sess, variable.On); err != nil {
 		return err
 	}
+	if err = setTiDBTTLJobEnable(d.ctx, sess, variable.Off); err != nil {
```
Comment (YangKeao) [inline ddl/cluster.go:183]: As we have turned off the `TTL_ENABLE` for every table, is it still necessary to turn off this variable?

Comment (lcwangchao) [reply ddl/cluster.go:192]: for flashback cluster，we did not turn off any table's TTL_ENABLE and just set `tidb_ttl_job_enable='OFF'`

File: ddl/cluster.go (hunk 1 above)
Comment (lcwangchao) [reply ddl/cluster.go:700]: done

**Review by Defined2014 — APPROVED** [review]

**Review by YangKeao — APPROVED** [review]

**Pull request merged**